big-o/big-o-notation.md

```diff
@@ -107,10 +122,128 @@ Consider this problem: You have an address book, and you want to find the contac
 
 <!-- Question about comparing Big Os for more efficient -->
 
-<!-- Question about comparing Big Os for more efficient -->
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: ordering
+* id: d2d156f6-aaab-4aba-9cf7-b9cdaceed013
+* title: Time/Space Complexities Worst-to-best
+* points: 1
+* topics: big-o
+
+##### !question
+
+Arrange the following complexities from worst to best.
+
+##### !end-question
+
+##### !answer
+
+1. O(1)
+1. O(2^n)
+1. O(n log n)
+1. O(log n)
+1. O(n^2)
+1. O(n)
```

Comment:
How does Learn know the order for this?  Is it supposed to be listed in the correct order and Learn will scramble it?  If so, I'm confused, since the problem says to list these from worst to best, but at either extremity we have O(1) and O(n), which are both pretty dang good...

Am I missing something?

Reply:
It should be listed in the correct order, Ansel's totally correct!